rollup/circuitcapacitychecker/libzkp/src/lib.rs

```diff
@@ -158,6 +121,25 @@ pub mod checker {
         };
         serde_json::to_vec(&r).map_or(null(), vec_to_c_char)
     }
+
+    unsafe fn apply_block_inner(id: u64, block_trace: *const c_char) -> Result<RowUsage, Error> {
+        log::debug!(
+            "ccc apply_block raw input, id: {:?}, block_trace: {:?}",
+            id,
+            c_char_to_str(block_trace)
+        );
+        let block_trace = c_char_to_vec(block_trace);
+        let traces = serde_json::from_slice::<BlockTrace>(&block_trace)
+            .unwrap_or_else(|_| panic!("id: {id:?}, fail to deserialize block_trace"));
+        CHECKERS
+            .get_mut()
+            .expect("fail to get circuit capacity checkers map in apply_block")
+            .get_mut(&id)
+            .unwrap_or_else(|| {
+                panic!("fail to get circuit capacity checker (id: {id:?}) in apply_block")
```

Comment:
why still panic?

Reply:
This panic is caused by `HashMap.Get(id)`, I think it will not panic when the program starts up successfully because we always set the ID into CHECKERS when `new_circuit_capacity_checker` at the beginning. 
So I think this panic will not appear once geth activated.

Reply:
not matter whether it will happen, we can throw error instead of panic.

The idea of PR is to return error instead of panic.
that mean you should not panic/assert/expect in your PR.

if it panic in `estimate_circuit_capacity` then it's not our bussiness